test/m_f_c_coin.js

```diff
@@ -18,5 +84,350 @@ contract('MFC_Token', function(accounts) {
+            [
+            owner,
+            receiver.address,
+            amount,
+            '0x00'
+            ]
+        );
+        await web3.eth.sendTransaction({from: spender, to: contract.address, data: transferFromMethodTransactionData, value: 0});
+
+        balance = await contract.balanceOf.call(receiver.address);
+        let receiver_balance_after = balance.toNumber();
+        balance = await contract.balanceOf.call(owner);
+        let owner_balance_after = balance.toNumber();
+
+        assert.equal(owner_balance_after, owner_balance_befor - amount, "Amount wasn't correctly taken from the sender");
+        assert.equal(receiver_balance_after, receiver_balance_befor + amount, "Amount wasn't correctly sent to the receiver");
+    });
+
+    /* if the sender have enough allowed tokens and send to contract without tokenFallback() */
+    it("owner should approve amount of tokens to spender and spender should try to transfer amount of tokens to receiver, where receiver is contract without ERC223Receiver interface and should receive an error: revert", async function() {
+        let owner = accounts[0];
+        let spender = accounts[11];
+        let amount = 100;
+
+        let contract = await token.deployed();
+        let receiver = await StandardToken.deployed();
+
+        let balance = await contract.balanceOf.call(receiver.contract.address);
+        let receiver_balance_befor = balance.toNumber();
+        balance = await contract.balanceOf.call(owner);
+        let owner_balance_befor = balance.toNumber();
+
+        await contract.approve(spender, amount, {'from': owner});
+        const transferFromMethodTransactionData = web3Abi.encodeFunctionCall(
+            overloadedTransferFromAbi,
+            [
+            owner,
+            receiver.address,
+            amount,
+            '0x00'
+            ]
+        );
+        try {
+            await web3.eth.sendTransaction({from: spender, to: contract.address, data: transferFromMethodTransactionData, value: 0});
+            assert.ifError('Error, previous code must throw exception');
+        } catch (err){
+            assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Wrong error");
+        };
+
+        balance = await contract.balanceOf.call(receiver.address);
+        let receiver_balance_after = balance.toNumber();
+        balance = await contract.balanceOf.call(owner);
+        let owner_balance_after = balance.toNumber();
+
+        assert.equal(owner_balance_after, owner_balance_befor, "Balance changed, it isn't correct");
+        assert.equal(receiver_balance_after, receiver_balance_befor, "Balance changed, it isn't correct");
+    });
+
+    /* if the sender have not enough allowed tokens and send to address */
+    it("owner should approve amount of tokens to spender and spender should try to transfer more then amount of tokens to receiver and should receive an error: revert", async function() {
```

Comment:
There is very big test description.
It is better to full case described in the comment.
And test name should be smth:
"test transferFrom with more amount than approved"

And in other places too